web/hooks/use-notifications.ts

```diff
@@ -19,31 +20,47 @@ export type NotificationGroup = {
 
 // For some reason react-query subscriptions don't actually listen for notifications
 // Use useUnseenPreferredNotificationGroups to listen for new notifications
-export function usePreferredGroupedNotifications(privateUser: PrivateUser) {
+export function usePreferredGroupedNotifications(
+  privateUser: PrivateUser,
+  cachedNotifications?: Notification[]
+) {
   const [notificationGroups, setNotificationGroups] = useState<
     NotificationGroup[] | undefined
   >(undefined)
-  const [notifications, setNotifications] = useState<Notification[]>([])
-  const key = `notifications-${privateUser.id}-all`
+  const [notifications, setNotifications] = useState<Notification[]>(
+    cachedNotifications ?? []
+  )
 
-  const result = useFirestoreQuery([key], getNotificationsQuery(privateUser.id))
-  useEffect(() => {
+  const result = useFirestoreQueryData(
+    ['notifications-all', privateUser.id],
+    getNotificationsQuery(privateUser.id)
+  )
+  useMemo(() => {
```

Comment:
Shouldn't this still be an effect if it's not returning a memo-ed data? Ditto for other uses of useMemo in this PR.